packages/web-config-server/src/apiV1/dashboards.js

```diff
@@ -3,33 +3,64 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
+import { filterEntities } from '@tupaia/utils';
 import orderBy from 'lodash.orderby';
 import { RouteHandler } from './RouteHandler';
 import { NoPermissionRequiredChecker } from './permissions';
 
+const NO_DATA_AT_LEVEL_DASHBOARD_ITEM_CODE = 'no_data_at_level';
+const NO_ACCESS_DASHBOARD_ITEM_CODE = 'no_access';
+
+const checkEntityAgainstConditions = (entity, conditions = {}) =>
+  filterEntities([entity], conditions).length === 1;
+
 export default class extends RouteHandler {
   static PermissionsChecker = NoPermissionRequiredChecker;
 
   buildResponse = async () => {
     const { entity, query } = this;
-    const { code: entityCode } = entity;
-    const permissionGroups = await this.req.getUserGroups(entityCode);
     const hierarchyId = await this.fetchHierarchyId();
     // Fetch dashboards
     const dashboards = await this.models.dashboard.getDashboards(entity, hierarchyId);
+    if (!dashboards.length) {
+      return this.getStaticDashboard(entity, NO_DATA_AT_LEVEL_DASHBOARD_ITEM_CODE);
+    }
+
+    const dashboardsWithItems = await this.getDashboardsWithItems(
+      dashboards,
+      entity,
+      query.projectCode,
+    );
+    const possibleDashboardRelations = await this.models.dashboardRelation.find({
+      dashboard_id: dashboards.map(d => d.id),
+    });
+
+    if (!dashboardsWithItems.length && possibleDashboardRelations.length) {
+      return this.getStaticDashboard(entity, NO_ACCESS_DASHBOARD_ITEM_CODE);
+    }
+
+    return dashboardsWithItems;
+  };
+
+  getDashboardsWithItems = async (dashboards, entity, projectCode) => {
+    const permissionGroups = await this.req.getUserGroups(entity.code);
     const sortedDashboards = orderBy(dashboards, ['sort_order', 'name']);
```

Comment:
Is it okay to sort all dashboards by name? The order will be different from how it is now in tupaia.org